### PR TITLE
K8s 1.26 upgrade

### PR DESCRIPTION
**Pull request by inFocus7**
- [X] upgrade k8s dependencies to 1.26
- [X] update go-utils to v0.24.2
- [x] Upgrade to a released `k8s-utils` version that has [1.26 bumps](https://github.com/solo-io/k8s-utils/pull/36)

### CODE REVIEW

**Comment by inFocus7** [comment]
tried generating code locally with `make clean generate-all -B`, but got an error
```
"Thu, 18 May 2023 09:58:36 EDT: github.com/solo-io/solo-kit/generate.go:37"     generate failed!: &errors.withStack{
  error: &errors.withMessage{
    cause: &exec.ExitError{
      ProcessState: &os.ProcessState{
        pid:    74978,
        status: 0x00000100,
        rusage: &syscall.Rusage{
          Utime: syscall.Timeval{
            Sec:       0,
            Usec:      6536,
            Pad_cgo_0: [4]uint8{
              0x00, 0x00, 0x00, 0x00,
            },
          },
          Stime: syscall.Timeval{
            Sec:       0,
            Usec:      3659,
            Pad_cgo_0: [4]uint8{
              0x00, 0x00, 0x00, 0x00,
            },
          },
          Maxrss:   6389760,
          Ixrss:    0,
          Idrss:    0,
          Isrss:    0,
          Minflt:   784,
          Majflt:   0,
          Nswap:    0,
          Inblock:  0,
          Oublock:  0,
          Msgsnd:   0,
          Msgrcv:   0,
          Nsignals: 0,
          Nvcsw:    3,
          Nivcsw:   4,
        },
      },
      Stderr: []uint8{},
    },
    msg: "[protoc -I/Users/fabiangonz98/go/src/github.com/solo-io/solo-kit/vendor_any -I/Users/fabiangonz98/go/src/github.com/solo-io/solo-kit/vendor_any/github.com/solo-io/protoc-gen-ext/external -I/Users/fabiangonz98/go/src/github.com/solo-io/solo-kit/vendor_any/github.com/envoyproxy/protoc-gen-validate -I/Users/fabiangonz98/go/src/github.com/solo-io/solo-kit/vendor_any/github.com/solo-io/protoc-gen-ext -I/Users/fabiangonz98/go/src/github.com/solo-io/solo-kit/vendor_any/github.com/solo-io/solo-kit/api/external --openapi_out=yaml=true,single_file=false,include_description=true,enum_as_int_or_string=false,additional_empty_schema=core.solo.io.Status:/var/folders/5_/gy4y2g551_dcsykch8xy07sh0000gn/T/549943686/mock_resources -o /var/folders/5_/gy4y2g551_dcsykch8xy07sh0000gn/T/schema-gen-560651527 --include_imports --include_source_info github.com/solo-io/solo-kit/test/mocks/api/v1/mock_resources.proto] failed: github.com/solo-io/solo-kit/test/mocks/api/v1/mock_resources.proto: warning: Import google/protobuf/any.proto but not used.\ngithub.com/solo-io/solo-kit/test/mocks/api/v1/mock_resources.proto: warning: Import google/protobuf/struct.proto but not used.\nprotoc-gen-openapi: program not found or is not executable\n--openapi_out: protoc-gen-openapi: Plugin failed with status code 1.\n",
  },
  stack: &errors.stack{
    0x1051615ec,
    0x10516157d,
    0x10525dc9c,
    0x10525d9b4,
    0x10525cad8,
    0x10529515c,
    0x105293eac,
    0x1052974b0,
    0x104a4bfa0,
    0x104a78ec4,
  },
}
exit status 1
```

**Comment by inFocus7** [comment]
Reverted my try to move tests to GH Actions + created an issue for it #543

File: go.mod [hunk 1]
```diff
@@ -20,54 +20,50 @@ require (
 	github.com/hashicorp/vault/api v1.0.5-0.20191108163347-bdd38fca2cff
 	github.com/iancoleman/strcase v0.0.0-20191112232945-16388991a334
 	github.com/mitchellh/hashstructure v1.0.0
-	github.com/onsi/ginkgo/v2 v2.5.0
+	github.com/onsi/ginkgo/v2 v2.6.0
 	github.com/onsi/gomega v1.24.1
 	github.com/pkg/errors v0.9.1
-	github.com/pseudomuto/protoc-gen-doc v1.5.1
+	github.com/pseudomuto/protoc-gen-doc v1.5.0
```
Comment (jenshu) [inline go.mod:26]: do you know why this got downgraded?

Comment (inFocus7) [reply go.mod:26]: Hmm not sure,  i'll check go-utils & k8s-utils to see which might've done it :hmm:

Comment (inFocus7) [reply go.mod:26]: Yeah not sure. looks like it was part of [this commit](https://github.com/solo-io/solo-kit/pull/542/commits/eb2501fe78e49efa241aaa3ae9871c5536a52d8e) where i bumps k8s-utils to a specific commit before release. I think I can `go get github.com/pseudomuto/protoc-gen-doc@v1.5.1` if you think that'd be safer

**Review by jenshu — APPROVED** [review]

**Pull request merged**